docs: add meta description (#51073)
LLNL/spack: lib/spack/docs/configuring_compilers.rst

   SPDX-License-Identifier: (Apache-2.0 OR MIT)

.. meta::
   :description lang=en:

>       Discover how to configure compilers in Spack, whether by specifying them as externals, installing them in the Spack store, or making them available through a buildcache.

Author: ```suggestion
      Discover how to configure compilers in Spack, whether by specifying them as externals, or by installing them with Spack.
```

.. code-block:: console

   $ spack install hdf5~mpi %gcc@14

> The same holds true for compilers that are made available from build caches, when reusing them is allowed.

Author: Not really related to this PR, but I've removed this line because at this point in the docs reuse concretization is not introduced, nor are build caches. It just begs the question "what build cache?" and what does "when reusing them is allowed" mean.

As far as I know, we don't provide compilers in build caches right now, with the exception of one gcc in the tutorial build cache.
